tests/integration-test/server/server.js

```diff
@@ -0,0 +1,410 @@
+'use strict';
+
+const Busboy = require('busboy');
+const debug = require('debug')('sdk-integration-server');
+const fs = require('fs');
+const path = require('path');
+const http = require('http');
+const url = require('url');
+
+const R = require('ramda');
+
+const server = (function genServer() {
+  const BASE_DIR = __dirname;
+  const PORT = 4000;
+  var TASK = {};
+  var eventList = {
+    getTaskJSON: false,
+    getInputFile: false,
+    running: false,
+    writeStatus: false,
+    numStatusWrites: 0,
+    writeLogfile: false,
+    numLogfileWrites: 0,
+    complete: false,
+    failed: false,
+    failureType: null,
+    writeOutput: false,
+    reportMetadata: false,
+    reportTransactionUnits: false,
+  };
+
+  const getQueryFilename = (qp) => qp.split('/').pop();
+  const handler = R.curry(resHandler);
+
+  const ROUTES = [
```

Comment:
I like this way of defining routes

Reply:
Could add middleware as an array of functions too.